daffodil-japi/build.sbt

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-addCompilerPlugin("com.typesafe.genjavadoc" %% "genjavadoc-plugin" % "0.11" cross CrossVersion.full)
+addCompilerPlugin("com.typesafe.genjavadoc" %% "genjavadoc-plugin" % "0.16" cross CrossVersion.full)
```

Comment:
Looks like something is wrong with our java docs now. GitHub actions are failing at generating the docs.

Reply:
So, our sbt publishLocal has no issues, so the options to the github actions are somehow different and less tolerant.

Reply:
I am reverifying that sbt publishLocal does not produce these errors when creating the doc jars.

Reply:
I'm not sure if publishLocal will generate javadocs. The command to do that is ``sbt daffodil-japi/genjavadoc:doc``. But that works fine on my machine. It's not clear to me what github actions is doing differently to break this.

Reply:
Confirmed. The sbt publishLocal constructs javadoc jars without error. So github actions is doing this differently than our sbt build does.

Reply:
I think the key difference is the "coverage". We run sbt coverage ... which changes things. I ran into this before.

Reply:
we have a whole build.sbt in daffodil-japi which is very much about the javadoc, though I don't see any specific options there. But it does explicitly request genjavadoc-plugin 0.16 now.

Reply:
Good call Lola, [DAFFODIL-2111](https://issues.apache.org/jira/browse/DAFFODIL-2111) is probably what's goin on here. Maybe we can just add ``coverageOff`` for the doc part of the test?
```
run: $SBT coverageOff daffodil-japi/genjavadoc:doc daffodil-sapi/doc
```

Reply:
The main.yml github action script just invokes daffodil-japi/genjavadoc:doc. 

When I do this manually it works fine. 

I have no clue where the difference in behavior comes from here.

Reply:
Ah.... It's because I hadn't updated the main.yml initially. I will push the updates where main.yml will have the right scala version number to go with the genjavadoc 0.16 and we'll see if it works then.

Reply:
The "coverage" comes from the SBT environment variable at the top of the main.yml. It's so that every command is run with coverage enabled to ensure all the compmilation and test steps have the run thing. We should verify that sbt won't try to recompile everything if we disable coverage for just the docs.

Reply:
I will add the coverageOff thing.

I also found this comment in doc on genjavadoc:

    On recent version of the JDK you will have to pass the --ignore-source-errors flag to make the javadoc tool accept our output.